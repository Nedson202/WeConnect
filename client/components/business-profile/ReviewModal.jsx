import React from 'react';
import PropTypes from 'prop-types';
import ReactStars from 'react-stars';

/**
 * @class ReviewModal
 *
 * @extends {Component}
 */
const ReviewModal = ({
  onChange, onStarClick, onSubmit, state, user, business
}) => {
  const { message, rating } = state;
  return (
    <div>
      { user.username === process.env.ADMIN ? null :
      <div>
        { business.userId === user.userId ? null :
        <div className="review-section row">
          <div className="col-lg-9">
            <div className="form-group">
              <label id="control-label">Add a review</label>
              <textarea
                value={message}
                onChange={onChange}
                type="text"
                name="message"
                className="form-control"
                id="control-label"
                placeholder="Add review"
                rows="3"
              />
            </div>
            <ReactStars
              count={5}
              onChange={onStarClick}
              size={19}
              color1="#333333"
              color2="#ffaf00"
              value={rating}
            />
          </div>
          <div className="col-lg-3 add-review">
            <button
              // disabled={isPosting}
              className="btn btn-outline-success add-review-button"
              id="add-review"
              onClick={onSubmit}
            >
              <span>Post review</span>
            </button>
          </div>
        </div>
        }
      </div>}
    </div>
  );
};

ReviewModal.propTypes = {
  state: PropTypes.object.isRequired,
  onStarClick: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
  business: PropTypes.object.isRequired
};

export default ReviewModal;
