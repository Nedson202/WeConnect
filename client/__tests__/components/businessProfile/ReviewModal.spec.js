import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import ReviewModal from '../../../components/business-profile/ReviewModal';
import businessData from '../../__mockData__/business';
import userData from '../../__mockData__/userData';

const { business, review, businessOwner } = businessData;
const { userUpdateData } = userData;
configure({ adapter: new Adapter() });

let props;
const setup = () => {
  props = {
    onChange: jest.fn(() => Promise.resolve()),
    onStarClick: jest.fn(() => Promise.resolve()),
    onSubmit: jest.fn(() => Promise.resolve()),
    state: {
      message: review.message,
      rating: review.rating,
      isLoading: false
    },
    user: userUpdateData,
    business
  };

  return shallow(<ReviewModal {...props} />);
};


describe('Component: ReviewModal', () => {
  beforeEach(() => {
    global.window = {
      window: () => {},
    };
  });

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(2);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });

  it('should render form if reviewer is not the business owner', () => {
    const wrapper = setup();
    wrapper.setProps({ business: businessOwner });
    expect(wrapper.find('div').length).toBe(6);
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('textarea').length).toBe(1);
    expect(wrapper.find('ReactStars').length).toBe(1);
    expect(wrapper.find('span').length).toBe(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });
});