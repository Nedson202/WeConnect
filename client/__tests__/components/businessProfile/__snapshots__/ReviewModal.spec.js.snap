// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ReviewModal should render 1`] = `
<div>
  <div />
</div>
`;

exports[`Component: ReviewModal should render form if reviewer is not the business owner 1`] = `
<div>
  <div>
    <div
      className="review-section row"
    >
      <div
        className="col-lg-9"
      >
        <div
          className="form-group"
        >
          <label
            id="control-label"
          >
            Add a review
          </label>
          <textarea
            className="form-control"
            id="control-label"
            name="message"
            onChange={[MockFunction]}
            placeholder="Add review"
            rows="3"
            type="text"
            value="Great business"
          />
        </div>
        <ReactStars
          char="â˜…"
          color1="#333333"
          color2="#ffaf00"
          count={5}
          edit={true}
          half={true}
          onChange={[MockFunction]}
          size={19}
          value={3}
        />
      </div>
      <div
        className="col-lg-3 add-review"
      >
        <button
          className="btn btn-outline-success add-review-button"
          disabled={false}
          id="add-review"
          onClick={[MockFunction]}
        >
          <span>
            Post review
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
